notesapi/v1/tests/test_views.py

```diff
@@ -546,6 +567,93 @@ def test_search_course(self):
         self.assertEqual(results['total'], 1)
         self.assertEqual(results['rows'][0]['text'], u'Third note')
 
+    def test_search_tag(self):
+        """
+        Tests searching for tags
+        """
+        self._create_annotation(text=u'First note', tags=[u'foo', u'bar'])
+        self._create_annotation(text=u'Another one', tags=[u'bar'])
+        self._create_annotation(text=u'A third note', tags=[u'bar', u'baz'])
+        self._create_annotation(text=u'One final note', tags=[])
+
+        results = self._get_search_results(text='Foo')
+        self.assertEqual(results['total'], 1)
+        self.assertEqual(results['rows'][0]['text'], 'First note')
+
+        results = self._get_search_results(text='bar')
+        self.assertEqual(results['total'], 3)
```

Comment:
I've added a helper to check the text of the results without respect to order. My thinking here is that order should have already been tested by the ordering tests. The other tests should succeed regardless of which backend is doing the searching.